lms/djangoapps/verify_student/models.py

```diff
@@ -1112,8 +1112,13 @@ def get_or_create_verification_checkpoint(cls, course_id, checkpoint_location):
         Returns:
             VerificationCheckpoint object if exists otherwise None
         """
-        checkpoint, __ = cls.objects.get_or_create(course_id=course_id, checkpoint_location=checkpoint_location)
-        return checkpoint
+        try:
+            with transaction.atomic():
+                checkpoint, __ = cls.objects.get_or_create(course_id=course_id, checkpoint_location=checkpoint_location)
+                return checkpoint
+        except IntegrityError:
+            # Record already exists - just return it.
+            return cls.objects.get(course_id=course_id, checkpoint_location=checkpoint_location)
```

Comment:
I agree - thanks for catching this.

A couple of options here:
* Change this to return `None` when get_or_create fails and let the higher-level code decide what to do (probably re-try). Also change the test accordingly. 
* Re-try within this method to send back a proper record, which some parts of the code expects.